examples/AspNetCore/Program.cs

```diff
@@ -14,67 +14,133 @@
 // limitations under the License.
 // </copyright>
 
-using System;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+using System.Reflection;
+using Examples.AspNetCore;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Instrumentation.AspNetCore;
 using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
 using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
 
-namespace Examples.AspNetCore
+var builder = WebApplication.CreateBuilder(args);
+
+var serviceName = "AspNetCoreExampleService";
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+
+builder.Services.AddSwaggerGen();
+
+// OpenTelemetry
+var assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "unknown";
+
+var resourceBuilder = ResourceBuilder.CreateDefault()
```

Comment:
the resourcebuilder is not consistently used -- tracing creates a separate one.. metrics uses this etc. I'd prefer to show consistent usage of these. (would be good to address this as separate PRs)